js/web/lib/wasm/jsep/webgpu/ops/reduce.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, GpuDataType, ProgramInfo, ProgramInfoLoader, ProgramMetadata} from '../types';
+
+import {createIndicesHelper, ShaderHelper} from './common';
+
+const validateInputs = (inputs: readonly TensorView[]): void => {
+  if (!inputs || inputs.length == 0 || inputs.length > 2) {
+    throw new Error('Reduce op requires 1 or 2 inputs.');
+  }
+
+  if (inputs[0].dataType !== DataType.float) {
+    throw new Error('Invalid input type.');
+  }
```

Comment:
Reduce* operators are registered with both float32 and int32 but we only support float here. either only register float32 or implement both float32 and int32 will fix this.

Reply:
since in the shader code generation below hardcoded datatype 'f32' is used, I assume the shader cannot work with int32. how about revert the registration of int32 in this PR? maybe in future we can support int32 in a separated PR.

Reply:
Removed code added for int32_t.